src/systemctl/systemctl.c

```diff
@@ -5977,15 +5987,11 @@ static int trivial_method(int argc, char *argv[], void *userdata) {
 
         polkit_agent_open_maybe();
 
-        method =
-                streq(argv[0], "clear-jobs")    ||
-                streq(argv[0], "cancel")        ? "ClearJobs" :
-                streq(argv[0], "reset-failed")  ? "ResetFailed" :
-                streq(argv[0], "halt")          ? "Halt" :
-                streq(argv[0], "reboot")        ? "Reboot" :
-                streq(argv[0], "kexec")         ? "KExec" :
-                streq(argv[0], "exit")          ? "Exit" :
-                             /* poweroff */       "PowerOff";
+        NULSTR_FOREACH_PAIR(verb, method, table)
+                if (streq(argv[0], verb))
+                        break;
```

Comment:
I'm not convinced that this is really better. The ternary things was compact and pretty easy to read and didn't need an assert at the end.

Reply:
i am pretty sure anything that uses more than one ternary op in a row is pretty much unreadable, in particular if it keeps repeateding `streq(argv[0], …` all over the place...

Also, I think the assert() is a good thing, not a bad thing. The fact we can't really add it to the ternary op thing is a malus not a bonus, since it hides issue when for some reason this function gets called with an argument that we didn't expect...

But it's probably not worth the discussion, will drop from the PR...